doc/SATABS.md

```diff
@@ -0,0 +1,696 @@
+We document two programs that try to achieve formal guarantees of the
+absence of specific problems: CBMC and SATABS. The algorithms
+implemented by CBMC and SATABS are complementary, and often, one tool is
+able to solve a problem that the other cannot solve.
+
+Both CBMC and SATABS are verification tools for ANSI-C/C++ programs.
+They verify array bounds (buffer overflows), pointer safety, exceptions
+and user-specified assertions. Both tools model integer arithmetic
+accurately, and are able to reason about machine-level artifacts such as
+integer overflow. CBMC and SATABS are therefore able to detect a class
+of bugs that has so far gone unnoticed by many other verification tools.
+This manual also covers some variants of CBMC, which includes HW-CBMC
+for
+\ref man_hard-soft-introduction "hardware/software co-verification".
+
+## Automatic Program Verification with SATABS
+
+In many cases, lightweight properties such as array bounds do not rely
+on the entire program. A large fraction of the program is *irrelevant*
+to the property. SATABS exploits this observation and computes an
+*abstraction* of the program in order to handle large amounts of code.
+
+In order to use SATABS it is not necessary to understand the abstraction
+refinement process. For the interested reader, a high-level introduction
+to abstraction refinement is provided
+\ref man_satabs-overview "here".
+We also provide
+\ref man_satabs-tutorials "tutorials on how to use SATABS".
+
+Just as CBMC, SATABS attempts to build counterexamples that refute the
+property. If such a counterexample is found, it is presented to the
+engineer to facilitate localization and repair of the program.
+
+\subsection man_install-satabs Installing SATABS
+
+### Requirements
+
+SATABS is available for Windows, i86 Linux, and MacOS X. SATABS requires
+a code pre-processing environment comprising of a suitable preprocessor
+and an a set of header files.
+
+1.  **Linux:** the preprocessor and the header files typically come with
+    a package called *gcc*, which must be installed prior to the
+    installation of SATABS.
+2.  **Windows:** The Windows version of SATABS requires the preprocessor
+    `cl.exe`, which is part of Visual Studio (including the free [Visual
+    Studio
+    Express](http://msdn.microsoft.com/vstudio/express/visualc/)).
+3.  **MacOS:** Install
+    [XCode](http://developer.apple.com/technologies/xcode.html) prior to
```

Comment:
Do we actually need full XCode installed for any of CBMC? Or do we just need the "XCode command line tools", which is a significantly more lightweight package to install? I think just the command line tools can be installed with 'xcode-select --install' ?